src/mbgl/text/quads.cpp

```diff
@@ -21,14 +21,19 @@ SymbolQuad getIconQuad(const Anchor& anchor,
                        const SymbolLayoutProperties::Evaluated& layout,
                        const float layoutTextSize,
                        const style::SymbolPlacementType placement, 
-                       const Shaping& shapedText) {
-    auto image = *shapedIcon.image();
+                       const Shaping& shapedText,
+                       const float pixelRatio) {
+    const SpriteAtlasElement& image = shapedIcon.image();
 
+    // TODO: all the border-based adjustments preserve existing behavior in rendering tests.
+    // However, I suspect the render tests themselves are incorrect, and the adjustments should
+    // all be removed.
     const float border = 1.0;
-    auto left = shapedIcon.left() - border;
-    auto right = left + image.pos.w / image.relativePixelRatio;
-    auto top = shapedIcon.top() - border;
-    auto bottom = top + image.pos.h / image.relativePixelRatio;
+
+    float left = shapedIcon.left() - border;
+    float right = left + image.size[0] + 2 * border * pixelRatio / image.pixelRatio;
+    float top = shapedIcon.top() - border;
+    float bottom = top + image.size[1] + 2 * border * pixelRatio / image.pixelRatio;
```

Comment:
I think some border-based adjustments are needed (https://github.com/mapbox/mapbox-gl-native/pull/9038#issuecomment-302761272) but multiplying border by the pixel ratio on one side and not on the other looks suspicious.